avocado/core/job.py

```diff
@@ -137,16 +137,16 @@ def __init__(self, args=None):
         #: only once, since they are read only and will be shared across all
         #: tests of a job.
         self.test_parameters = None
-        if "test_parameters" in self.args:
+        if "test_parameters" in self.config:
             self.test_parameters = {}
-            for parameter_name, parameter_value in self.args.test_parameters:
+            for parameter_name, parameter_value in self.config.get('test_parameters'):
```

Comment:
I think it needs to iterate over `items()`.

Reply:
This is actually a list of tupples (see avocado/plugin/run.py:Run._test_parameters).